llvm/include/llvm/Transforms/Utils/LoopMetadata.h

```diff
@@ -0,0 +1,87 @@
+//===-- LoopMetadata.h - Convert Loop Metadata to assumptions --*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_TRANSFORMS_UTILS_LOOPMETADATA_H
+#define LLVM_TRANSFORMS_UTILS_LOOPMETADATA_H
+#include "llvm/IR/PassManager.h"
+#include "llvm/Passes/PassBuilder.h"
+
+namespace llvm {
+
+class Loop;
+/// Converts Loop Metadata to assumptions.
+class LoopMetadata : public PassInfoMixin<LoopMetadata> {
+private:
+  LLVMContext *Context;
+  ScalarEvolution *SE;
+  AssumptionCache *AC;
+  DominatorTree *DT;
+  const Loop *L;
+
+  unsigned MinIterCount;
```

Comment:
As we use iteration range as signed in the frontend and in the target, it is better to use signed here as well.

Reply:
went with signed